### PR TITLE
Add minor mode, keybindings and fix issue where counsel-gtags-pop would not return to original location

### PR DESCRIPTION
**Pull request by alexmurray**

### CODE REVIEW

File: counsel-gtags.el [hunk 1]
```diff
@@ -269,16 +296,61 @@
...
+Generate new TAG file in selected directory with `C-u C-u'"
+  (interactive)
+  (let ((how-to (counsel-gtags--how-to-update-tags))
+        (interactive-p (called-interactively-p 'interactive))
+        (current-time (float-time (current-time))))
+    (when (counsel-gtags--update-tags-p how-to interactive-p current-time)
+      (let* ((cmds (counsel-gtags--update-tags-command how-to))
+             (proc (apply #'start-file-process "counsel-gtags-update-tag" nil cmds)))
+        (if (not proc)
+            (message "Failed: %s" (string-join cmds " "))
```
Comment (syohex) [inline counsel-gtags.el]: Please use `mapconcat` instead of `string-join`. `string-join` was introduced at Emacs 24.4

Comment (alexmurray) [reply counsel-gtags.el]: No worries - have just pushed an extra commit for this https://github.com/syohex/emacs-counsel-gtags/pull/1/commits/e9b709c8981620b0142c17b320fe8982edf28e1b

**Comment by syohex** [comment]
Thanks

**Comment by alexmurray** [comment]
Do you have plans to add this to MELPA? I think it would be a great addition.

**Comment by syohex** [comment]
I thinks some updates are necessary to register to MELPA. I'll send request after it.

**Comment by alexmurray** [comment]
Anything I can help with?

**Comment by syohex** [comment]
Thanks, however I'll fix by myself.

**Comment by alexmurray** [comment]
No problem - it's your package after all :smile:

**Comment by syohex** [comment]
counsel-gtags have been registered in MELPA.
Sorry for too late my work.

**Pull request merged**